File: server/test/java/com.bomberman/TestDataGenerator.java
package com.bomberman;

import com.bomberman.constants.Configuration;
import com.bomberman.entities.Position;
import com.bomberman.entities.mapobject.Wall;
import com.bomberman.entities.mapobject.movable.Player;
import com.bomberman.exception.PlayerNotFoundException;
import com.bomberman.game.GameState;
import java.security.InvalidParameterException;

public class TestDataGenerator {
  private final GameState gameState;

  private TestDataGenerator() {
    gameState = new GameState();
  }

  public GameState getGameState() {
    return gameState;
  }

  public Player getPlayer(final String playerId) {
    return gameState.getPlayers().stream()
        .filter(player -> player.getId().equals(playerId))
        .findFirst()
        .orElseThrow(PlayerNotFoundException::new);
  }

  public static Position positionAtColRow(final int col, final int row) {
    return new Position(col * Configuration.TEXTURE_SIZE, row * Configuration.TEXTURE_SIZE);
  }

  public static TestDataGenerator newGenerator() {
    return new TestDataGenerator();
  }

  public TestDataGenerator createPlayerAtColRow(
      final String playerId, final int col, final int row) {
    return createPlayerAtPosition(
        playerId, col * Configuration.TEXTURE_SIZE, row * Configuration.TEXTURE_SIZE);
  }

  public TestDataGenerator createPlayerAtPosition(
      final String playerId, final float x, final float y) {
    checkBoundaries(x, y);
    gameState.getPlayers().add(new Player(playerId, new Position(x, y), gameState));
    return this;
  }

  public TestDataGenerator createWallAtColRow(final int col, final int row) {
    return createWallAtPosition(col * Configuration.TEXTURE_SIZE, row * Configuration.TEXTURE_SIZE);
  }

  public TestDataGenerator createWallAtPosition(final float x, final float y) {
    checkBoundaries(x, y);
    gameState.getWalls().add(new Wall(new Position(x, y)));
    return this;
  }

  private void checkBoundaries(final float x, final float y) {
    if (x > Configuration.MAP_WIDTH) {
      throw new InvalidParameterException(
          String.format("Position X can't be more than: %d", Configuration.MAP_WIDTH));
    }
    if (y > Configuration.MAP_HEIGHT) {
      throw new InvalidParameterException(
          String.format("Position Y can't be more than: %d", Configuration.MAP_HEIGHT));
    }
  }
}

File: server/src/test/kotlin/com/bomberman/TestDataGenerator.kt
package com.bomberman

import com.bomberman.entity.Position
import com.bomberman.entity.mapobject.Wall
import com.bomberman.entity.mapobject.movable.Player
import com.bomberman.game.GameState
import java.security.InvalidParameterException

class TestDataGenerator(
    val gameState: GameState = GameState()
) {

  fun getPlayer(playerId: String): Player =
      gameState.players.first { it.id == playerId }

  fun createPlayerAtColRow(playerId: String, col: Int, row: Int): TestDataGenerator =
      createPlayerAtPosition(playerId, col * TEXTURE_SIZE, row * TEXTURE_SIZE)

  fun createPlayerAtPosition(playerId: String, x: Int, y: Int): TestDataGenerator =
      createPlayerAtPosition(playerId, x.toFloat(), y.toFloat())

  fun createPlayerAtPosition(playerId: String, x: Float, y: Float): TestDataGenerator {
    checkBoundaries(x, y);
    gameState.players += Player(
        id = playerId,
        position = Position(x, y),
        gameState = gameState
    )
    return this
  }

  fun createWallAtColRow(col: Int, row: Int): TestDataGenerator {
    return createWallAtPosition(col * TEXTURE_SIZE, row * TEXTURE_SIZE)
  }

  fun createWallAtPosition(x: Int, y: Int): TestDataGenerator =
      createWallAtPosition(x.toFloat(), y.toFloat())

  fun createWallAtPosition(x: Float, y: Float): TestDataGenerator {
    checkBoundaries(x, y)
    gameState.walls += Wall(Position(x, y))
    return this
  }

  private fun checkBoundaries(x: Float, y: Float) {
    if (x > MAP_WIDTH) {
      throw InvalidParameterException("Position X can't be more than: $MAP_WIDTH")
    }
    if (y > MAP_HEIGHT) {
      throw InvalidParameterException("Position Y can't be more than $MAP_HEIGHT")
    }
  }
}

fun positionAtColRow(col: Int, row: Int): Position =
    Position(col * TEXTURE_SIZE, row * TEXTURE_SIZE)